### PR TITLE
grid/license-key-expired-data

### PR DESCRIPTION
**Pull request by sebastianbochan**
Added `gridKey` to grid-pro.

-------
- [x] - sync Master branch
- [x] - 3 tests - 3 keys
- [x] - composition + events
- [x] - Decode
	- [x] - decode date
	- [x] - checksum
- [x] - Error handling (console.warn)
- [x] - Cleanup code
- [x] - update edge cases (double wrong keys, but different status).
- [x] - tests
   - [x] - checksum issue 
- [x] - api docs article - https://github.com/highcharts/highcharts/pull/24489

### CODE REVIEW

**Review by skrulling — APPROVED** [review]
Great stuff 🙌

File: ts/Grid/Pro/License/LicenseValidation.ts [hunk 1]
```diff
@@ -10,151 +10,365 @@
...
- * Validates Grid Pro Grid Keys using checksum algorithm.
+ * Result of status for a Grid Key string.
  * @internal
  */
+export enum LicenseStatus {
+    VALID = 'valid',
+    INVALID = 'invalid',
+    MISSING = 'missing',
+    EXPIRED = 'expired'
+}
```
Comment (ddragula) [inline ts/Grid/Pro/License/LicenseValidation.ts]: use `const licenseStatus = { ... } as const`; instead of enum

File: ts/Grid/Pro/License/LicenseValidation.ts [hunk 2]
```diff
@@ -10,151 +10,365 @@
...
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The class that handles Grid Pro license keys: validation on load and update,
+ * parsing, and status checks for the grid.
+ */
 class LicenseValidation {
```
Comment (ddragula) [inline ts/Grid/Pro/License/LicenseValidation.ts]: Use named module instead of class, since all the methods and fields are static anyway.

**Review by ddragula — CHANGES_REQUESTED** [review]

**Review by stitot — APPROVED** [review]

File: test/ts-node-unit-tests/tests/GridLicenseValidation.test.ts [hunk 3]
```diff
@@ -1,23 +1,86 @@
...
+/**
+ * Fixed `today` for annual tests only: `getStatus` uses `new Date()` for
+ * non-perpetual keys.
+ */
+const STATIC_DAY = new Date(Date.UTC(2026, 3, 13));
+
+const KEY_ANNUAL_VALID = 'NMD7-4JNR-GU9L-A223-06PP-0000';
+const KEY_ANNUAL_EXPIRED = 'C2TR-Z9ZC-OL4D-A1HT-07CF-0000';
+/** Perpetual expiry date: 2025-10-21 */
+const KEY_PERPETUAL_SUPPORT_ENDED = '5DR9-W35I-TMXI-P1MW-07PU-0000';
```
Comment (stitot) [inline test/ts-node-unit-tests/tests/GridLicenseValidation.test.ts]: I guess these will be added to our public repo? Is that wise?

**Review by stitot — APPROVED** [review]

**Review by ddragula — APPROVED** [review]
Approved as the requested changes are provided in this PR: https://github.com/highcharts/highcharts/pull/24523
